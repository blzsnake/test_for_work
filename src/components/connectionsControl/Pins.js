/* eslint-disable camelcase */
import React from 'react';
import { Text } from 'ghofi-react-components';

import s from './styles.less';

/**
 * @param  {} {pins=[]}
 */

const Pins = ({ pins = [], onPinClick, client_id }) => (
  <ul className={s.pins} >
    {
      pins.map(item => (
        <li
          className={s.pins__item}
          onKeyDown={onPinClick({ pin: String(item), client_id, name: '', mac: '', type: '' })}
          onClick={onPinClick({ pin: String(item), client_id, name: '', mac: '', type: '' })}
          // eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
          role='button'
          tabIndex={0} >
          <Text
            size={16}
            sizeTablet={14}
            uppercase >
            {item}
          </Text>
        </li>))
    }
    {/* <li className={s.pins__add}>
      <Button
        wide
        color='yellow'>
        <Text
          size={14}
          fontInherit
          uppercase >
          <FormattedMessage id='connections_control_pins' />
        </Text>
      </Button>
    </li> */}
  </ul>
);

export default Pins;
