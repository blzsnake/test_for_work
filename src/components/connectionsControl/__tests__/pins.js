import React from 'react';
import { shallow } from 'enzyme';

import Pins from '../Pins';

const props = {
  pins: ['string'],
  onPinClick: () => {},
};

describe('Should render', () => {
  it('Pins render', () => {
    const component = shallow(<Pins {...props} />);

    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
