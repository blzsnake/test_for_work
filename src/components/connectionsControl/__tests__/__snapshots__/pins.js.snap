// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Pins render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pins
    onPinClick={[Function]}
    pins={
      Array [
        "string",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className={undefined}
          onClick={undefined}
          onKeyDown={undefined}
          role="button"
          tabIndex={0}
        >
          <Styled(Text)
            size={16}
            sizeTablet={14}
            uppercase={true}
          >
            string
          </Styled(Text)>
        </li>,
      ],
      "className": "pins",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Styled(Text)
            size={16}
            sizeTablet={14}
            uppercase={true}
          >
            string
          </Styled(Text)>,
          "className": undefined,
          "onClick": undefined,
          "onKeyDown": undefined,
          "role": "button",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "string",
            "size": 16,
            "sizeTablet": 14,
            "uppercase": true,
          },
          "ref": null,
          "rendered": "string",
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className={undefined}
            onClick={undefined}
            onKeyDown={undefined}
            role="button"
            tabIndex={0}
          >
            <Styled(Text)
              size={16}
              sizeTablet={14}
              uppercase={true}
            >
              string
            </Styled(Text)>
          </li>,
        ],
        "className": "pins",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Styled(Text)
              size={16}
              sizeTablet={14}
              uppercase={true}
            >
              string
            </Styled(Text)>,
            "className": undefined,
            "onClick": undefined,
            "onKeyDown": undefined,
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "string",
              "size": 16,
              "sizeTablet": 14,
              "uppercase": true,
            },
            "ref": null,
            "rendered": "string",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
